src/mgr/MgrClient.cc

```diff
@@ -391,6 +398,11 @@ bool MgrClient::handle_mgr_configure(MMgrConfigure *m)
     _send_stats();
   }
 
+  if (m->query_list.size()) /* if there's a query, send it */ 
+  {
+    send_perf_metric_query();
```

Comment:
I think it is not what we want her (`handle_mgr_configure`). On receiving a configure message we want to register this query in the osd so it start/stop to collect dynamic perf stats. And to send this stats we want in `_send_stats()`, together with other osd stats.

Reply:
Good point -- that would change ```send_perf_metric_query``` to ```set_perf_metric_query``` (or similar) and the callback to the OSD would be changed from ```MMgrReport* OSD::collect_perf_metrics()``` to ```void OSD::set_perf_metric_query(bool enabled)``` (temporary param for now).

Reply:
... and you would need a second OSD callback (via ```set_perf_metric_query_cb```) to populate the results in the provided ```MMgrReport*``` message

Reply:
yes, and I had to do this in my #24265. So it can be changed similarly here, or any other way and I will update my commit, or just leave as it is now so it is fixed in my PR.

Reply:
I tried to address this and the grouping of the `set_[sg]et_perf_*_cb()` (as mentioned in #24265 earlier) in commits https://github.com/ceph/ceph/pull/24180/commits/0bef65f3ed1cedf87ebb8c6ae57167f06400ffc5 and https://github.com/ceph/ceph/pull/24180/commits/141524b7e8617398a6c5e6023e0df9f2860f917d. 

Is that what you expected ? I tried to keep consistent with what is in #24265 but let me know in case I forgot anything.

Reply:
Yeah -- thx.